lib/java/com/google/android/material/bottomsheet/BottomSheetBehavior.java

```diff
@@ -288,10 +288,12 @@ public boolean onInterceptTouchEvent(CoordinatorLayout parent, V child, MotionEv
       case MotionEvent.ACTION_DOWN:
         int initialX = (int) event.getX();
         initialY = (int) event.getY();
-        View scroll = nestedScrollingChildRef != null ? nestedScrollingChildRef.get() : null;
-        if (scroll != null && parent.isPointInChildBounds(scroll, initialX, initialY)) {
-          activePointerId = event.getPointerId(event.getActionIndex());
-          touchingScrollingChild = true;
+        if (state != STATE_SETTLING) {
```

Comment:
Does this mean we won't intercept touch events while the view is settling? What if the user wants to stop the current settle animation to move the BottomSheet?

Reply:
It does not mean we won't intercept touch event.
This code block is **for scrolling of nested child**, not for dragging the bottom sheet.
Dragging events will be processed by `viewDragHelper`, not by here.

As you mentioned, when user wanted to stop the current settle animation, it was not worked if the nested scroll child is existed, and the nested scroll child was scrolled instead.
So I fixed it with this CL.

When `touchingScrollingChild` is set `true`, `viewDragHelper` will **not** capture the view that should be dragged, and then, nested scrolling will be performed.
So, we have to leave `touchingScrollingChild` as `false` to disable nested scrolling in settling state.

Reply:
Thanks for the explanation. That makes sense 👍